src/test/operators/join_multi_predicates_test.cpp

```diff
@@ -0,0 +1,359 @@
+// A1 B4 C2 D3 E1 F2 G1 H1 I1       InnerLTableSmallerRTableRandomNullsEqLt
+// A1 B6 C2 D3 E1 F2 G1 H1 I1       InnerLTableSmallerRTableRandomNullsEqNe
+// C variations:
+// A1 B5 C1 D3 E1 F2 G1 H1 I1       InnerLTableSmallerRTableRandomNullsEqGt
+// A1 B5 C3 D3 E1 F2 G1 H1 I1       InnerLTableSmallerRTableRandomNullsEqGt
+// A1 B5 C4 D3 E1 F2 G1 H1 I1       InnerLTableSmallerRTableRandomNullsEqGt
+// D variations:
+// A1 B5 C2 D1 E1 F2 G1 H1 I1       InnerLTableSmallerRTableRandomNullsEqGt
+// A1 B5 C2 D2 E1 F2 G1 H1 I1       InnerLTableSmallerRTableRandomNullsEqGt
+// A1 B5 C2 D4 E1 F2 G1 H1 I1       InnerLTableSmallerRTableRandomNullsEqGt
+// E variations:
+// A1 B5 C2 D3 E2 F2 G1 H1 I1       InnerLTableSmallerRTableNullsLastEqGt
+// A1 B5 C2 D3 E3 F2 G1 H1 I1       InnerLTableSmallerRTableNullsFirstEqGt
+// F variations:
+// A1 B5 C2 D3 E1 F1 G1 H1 I1       InnerLTableLargerRTableRandomNullsEqGt
+// A1 B5 C2 D3 E1 F3 G1 H1 I1       InnerLTableSameSizeRTableRandomNullsEqGt
+// G variations:
+// A1 B5 C2 D3 E1 F2 G2 H1 I1       InnerLTableSmallerRTableRandomNullsEqGtEq
+// H variations:
+// A1 B5 C2 D3 E1 F2 G1 H2 I1       InnerLTableSmallerRTableRandomNullsDifferentDataTypesEqGt
+// I variations:
+// A1 B5 C2 D3 E1 F2 G1 H1 I2       InnerLTableSmallerRTableRandomNullsStringComparisonEqGt
+
+// TODO(anyone) add other Join types when they support multi predicates
+using JoinMultiPredicateTypes = ::testing::Types<JoinHash>;
+TYPED_TEST_CASE(JoinMultiPredicateTest, JoinMultiPredicateTypes, );  // NOLINT(whitespace/parens)
+
+TYPED_TEST(JoinMultiPredicateTest, InnerLTableSmallerRTableRandomNullsEqGt) {
+  this->_test_join_output(this->_base_choice_join_parameters.value());
+}
+
+TYPED_TEST(JoinMultiPredicateTest, LeftLTableSmallerRTableRandomNullsEqGt) {
+  auto parameters = this->_base_choice_join_parameters.value();
+  parameters.join_mode = JoinMode::Left;
+  parameters.expected_result_table_file_path =
+      "resources/test_data/tbl/joinoperators/multi_predicates/"
+      "result_left_a_nulls_random_b_nulls_random_larger_eq_gt.tbl";
+  this->_test_join_output(parameters);
+}
+
+TYPED_TEST(JoinMultiPredicateTest, RightLTableSmallerRTableRandomNullsEqGt) {
+  auto parameters = this->_base_choice_join_parameters.value();
+  parameters.join_mode = JoinMode::Right;
+  parameters.expected_result_table_file_path =
+      "resources/test_data/tbl/joinoperators/multi_predicates/"
+      "result_right_a_nulls_random_b_nulls_random_larger_eq_gt.tbl";
+  this->_test_join_output(parameters);
+}
+
+TYPED_TEST(JoinMultiPredicateTest, OuterLTableSmallerRTableRandomNullsEqGt) {
+  auto parameters = this->_base_choice_join_parameters.value();
+  parameters.join_mode = JoinMode::Outer;
+  parameters.expected_result_table_file_path =
+      "resources/test_data/tbl/joinoperators/multi_predicates/"
+      "result_outer_a_nulls_random_b_nulls_random_larger_eq_gt.tbl";
+  this->_test_join_output(parameters);
+}
+
+// test case is in conflict with
```

Comment:
what is the state on this? Will you fix the problem you referenced or does an issue need to be created for it?